addons/payment_razorpay/models/payment.py

```diff
@@ -0,0 +1,106 @@
+        self.ensure_one()
+        currency = self.env['res.currency'].sudo().browse(values['currency_id'])
+        if currency != self.env.ref('base.INR'):
+            raise ValidationError(_('Currency not supported by Razorpay'))
+        values.update({
+            'key': self.razorpay_key_id,
+            'amount': float_repr(float_round(values.get('amount'), 2) * 100, 0),
+            'name': values.get('partner_name'),
+            'contact': values.get('partner_phone'),
+            'email': values.get('partner_email'),
+            'order_id': values.get('reference'),
+        })
+        return values
+
+
+class PaymentTransaction(models.Model):
+    _inherit = 'payment.transaction'
+
+    @api.model
+    def _create_razorpay_capture(self, data):
+        payment_acquirer = self.env['payment.acquirer'].search([('provider', '=', 'razorpay')], limit=1)
+        payment_url = "https://%s:%s@api.razorpay.com/v1/payments/%s" % (payment_acquirer.razorpay_key_id, payment_acquirer.razorpay_key_secret, data.get('payment_id'))
+        try:
+            payment_response = requests.get(payment_url)
+            payment_response = payment_response.json()
+        except Exception as e:
+            raise e
+        reference = payment_response.get('notes', {}).get('order_id', False)
+        if reference:
+            transaction = self.search([('reference', '=', reference)])
+            capture_url = "https://%s:%s@api.razorpay.com/v1/payments/%s/capture" % (payment_acquirer.razorpay_key_id, payment_acquirer.razorpay_key_secret, data.get('payment_id'))
+            charge_data = {'amount': int(transaction.amount * 100)}
+            try:
+                payment_response = requests.post(capture_url, data=charge_data)
+                payment_response = payment_response.json()
+            except Exception as e:
+                raise e
+        return payment_response
+
+    @api.model
+    def _razorpay_form_get_tx_from_data(self, data):
+        reference, txn_id = data.get('notes', {}).get('order_id'), data.get('id')
+        if not reference or not txn_id:
+            error_msg = _('Razorpay: received data with missing reference (%s) or txn_id (%s)') % (reference, txn_id)
+            _logger.info(error_msg)
+            raise ValidationError(error_msg)
+
+        txs = self.env['payment.transaction'].search([('reference', '=', reference)])
+        if not txs or len(txs) > 1:
+            error_msg = 'Razorpay: received data for reference %s' % (reference)
+            if not txs:
+                error_msg += '; no order found'
+            else:
+                error_msg += '; multiple order found'
+            _logger.info(error_msg)
+            raise ValidationError(error_msg)
+        return txs[0]
+
+    @api.multi
+    def _razorpay_form_get_invalid_parameters(self, data):
+        invalid_parameters = []
+        if float_compare(float(data.get('amount', '0.0')) / 100, self.amount, precision_digits=2) != 0:
```

Comment:
you no longer need to cast to float in pytyon3, `1/2` gives 0.5 now (and for p2, an easier way is to divide by `100.0`)